nemo/collections/asr/parts/collections.py

```diff
@@ -289,7 +290,8 @@ def __init__(self, manifests_files: Union[str, List[str]], *args, **kwargs):
         for item in manifest.item_iter(manifests_files, parse_func=self.__parse_item):
             audio_files.append(item['audio_file'])
             durations.append(item['duration'])
-            labels.append(item['label'])
+            if not is_regression_task:
+                labels.append(item['label'])
```

Comment:
This is problematic. Your earlier code calls `sample.label` in the `_AudioLabelDataset` class - even for regression ... but `sample` is built from this list of values inside `item` - ie since you dont add `label` here for `is_regression_task=True`, `sample.label` raises an exception.

You _must pass ground truth value in the manifest_ - whether its called `label` or `target` or `value` makes little difference, but it must be added here. I would suggest adding the ground truth regression value here as just the `label` - but for regression case, explicitly cast the value to float.

So basically : 
```python
else:
    labels.append(float(item['label']))
```

Reply:
You are right about this. I've missed that because it was working anyway, and the reason is that the flag is_regression_task is not being populated to  `get_classification_label_dataset`.

I'm fixing this problem by adding the flag to the data loader config in the method `_setup_dataloader_from_config` in the class `_EncDecBaseModel,` what do you think?